modules/contrib/cyber_bridge/BUILD

```diff
@@ -0,0 +1,46 @@
+load("//tools:cpplint.bzl", "cpplint")
+
+package(default_visibility = ["//visibility:public"])
+
+cc_binary(
+    name = "cyber_bridge",
+    srcs = [
+        "bridge.cc",
+    ],
+    linkopts = [
+        "-lboost_system",
+    ],
+    deps = [
+        ":bridge_client",
+        ":bridge_server",
+        "//external:gflags",
+    ],
+)
+
+cc_library(
+    name = "bridge_server",
+    srcs = ["server.cc"],
+    hdrs = ["server.h"],
+    deps = [
+        ":bridge_client",
+    ],
+)
+
+cc_library(
+    name = "bridge_client",
+    srcs = [
+        "client.cc",
+        "clients.cc",
+        "node.cc",
```

Comment:
Please continue decoupling. In most cases we want at most 1 file in `srcs` and 1 in `hdrs`. Thanks!

Reply:
How do you deal with circular reference? I.e. class A uses class B, and class B uses class A too. In this case, do we have to put A.cc and B.cc in one build target?

Reply:
Actually one of the benefit of slim-target-principle is to avoid circular reference, which indicates some refactoring.
Or maybe you can try forward-declaration, while it's still not that good though.

Reply:
I separated out client.h, and now there are no more than 1 file in `srcs` and/or `hdrs`. The dependency graph is like the following:

cyber_bridge
 |
 |--bridge_server
     |
     |--bridge_client
         |
         |--bridge_clients
         |   |
         |   |--bridge_client_header
         |
         |--bridge_node
             |
             |--bridge_client_header